projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/datajobs/DeploymentModelConverter.java

```diff
@@ -183,4 +182,97 @@ private static DataJobDeploymentResources toDataJooDeploymentResources(
+    if (jobDeployment.getResources() != null) {
+      var resources = new DataJobDeploymentResources();
+      resources.setCpuRequestCores(jobDeployment.getResources().getCpuRequest());
+      resources.setCpuLimitCores(jobDeployment.getResources().getCpuLimit());
+      resources.setMemoryRequestMi(jobDeployment.getResources().getMemoryRequest());
+      resources.setMemoryLimitMi(jobDeployment.getResources().getMemoryLimit());
+      newDeployment.setResources(resources);
+    }
+    newDeployment.setLastDeployedBy(userDeployer);
+    newDeployment.setEnabled(jobDeployment.getEnabled());
+    newDeployment.setSchedule(jobDeployment.getSchedule());
+    return newDeployment;
+  }
+
+  public static DesiredDataJobDeployment mergeDeployments(
+      ActualDataJobDeployment oldDeployment, JobDeployment newDeployment, String userDeployer) {
+    if (oldDeployment.getDataJobName() == null
+        || newDeployment.getDataJobName() == null
+        || !oldDeployment.getDataJobName().equals(newDeployment.getDataJobName())
+        || !oldDeployment
+            .getDataJob()
+            .getJobConfig()
+            .getTeam()
+            .equals(newDeployment.getDataJobTeam())) {
+      throw new IllegalArgumentException(
+          "Cannot merge 2 deployments if team or job name is different."
+              + oldDeployment
+              + " vs "
+              + newDeployment);
+    }
+
+    DesiredDataJobDeployment mergedDeployment = new DesiredDataJobDeployment();
+    mergedDeployment.setDataJobName(newDeployment.getDataJobName());
+    mergedDeployment.setDataJob(oldDeployment.getDataJob());
+
+    mergedDeployment.setGitCommitSha(
+        (newDeployment.getGitCommitSha() != null
+            ? newDeployment.getGitCommitSha()
+            : oldDeployment.getGitCommitSha()));
+    mergedDeployment.setPythonVersion(
+        newDeployment.getPythonVersion() != null
+            ? newDeployment.getPythonVersion()
+            : oldDeployment.getPythonVersion());
+    mergedDeployment.setLastDeployedBy(
+        userDeployer != null ? userDeployer : oldDeployment.getLastDeployedBy());
+    mergedDeployment.setSchedule(
+        newDeployment.getSchedule() != null
+            ? newDeployment.getSchedule()
+            : oldDeployment.getSchedule());
+
+    mergeDeploymentResources(mergedDeployment, newDeployment, oldDeployment);
+
+    mergedDeployment.setEnabled(
+        newDeployment.getEnabled() != null
+            ? newDeployment.getEnabled()
+            : oldDeployment.getEnabled());
+
+    return mergedDeployment;
+  }
+
+  private static void mergeDeploymentResources(
+      DesiredDataJobDeployment mergedDeployment,
+      JobDeployment newDeployment,
+      ActualDataJobDeployment oldDeployment) {
+    if (newDeployment.getResources() == null) {
+      return;
+    }
+    DataJobDeploymentResources resources = new DataJobDeploymentResources();
+    resources.setCpuRequestCores(
```

Comment:
Can't we directly use thew newDeployment's resources instead of all these checks?
If that's not possible, do null checks for the resources of oldDeployment.

Reply:
In the current implementation when we patch two resources we only patch provided fields, . That's why we use these checks. E.g documentation in the current class - https://github.com/vmware/versatile-data-kit/blob/371a5188185b65ed0245f96e0a943c62daf3ffd7/projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentService.java#L58-L65

Reply:
Changing this to resolved, as we want to keep the existing logic of updating the existing deployment with any changed values from the new deployment.